diff --git a/node_modules/laravel-echo/src/util/index.ts b/node_modules/laravel-echo/src/util/index.ts
index e26a33e..4550f3e 100644
--- a/node_modules/laravel-echo/src/util/index.ts
+++ b/node_modules/laravel-echo/src/util/index.ts
@@ -1,16 +1,11 @@
 function isConstructor(obj: unknown): obj is new (...args: any[]) => any {
     try {
-        new (obj as new (...args: any[]) => any)();
-    } catch (err) {
-        if (
-            err instanceof Error &&
-            err.message.includes("is not a constructor")
-        ) {
-            return false;
-        }
+        // Avoid side effects when instantiating connector classes...
+        Reflect.construct(String, [], obj as new (...args: any[]) => any);
+        return true;
+    } catch {
+        return false;
     }
-
-    return true;
 }
 
 export { isConstructor };
